client/Cargo.toml

```diff
@@ -31,6 +31,9 @@ rand = "0.8"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 tungstenite = { version = "0.16" }
+smallvec = { version = "1.6.1", features = ["serde", "union"] }
+smallstr = { version = "0.2.0", features = ["serde", "union"] }
```

Comment:
why use these in client? is there a need to optimize the client?

Reply:
It does help the tests and both are indirect imports, so why not.

Reply:
I wouldn't like to bother the user with this. As long as it's not evident in the API I'm fine with this.